src/language/mongoDBService.ts

```diff
@@ -557,6 +595,86 @@ export default class MongoDBService {
     return [];
   }
 
+  // Highlight the usage of commands that only works inside interactive session.
+  provideDiagnostics(textFromEditor: string) {
+    const lines = textFromEditor.split(/\r?\n/g);
+    const diagnostics: Diagnostic[] = [];
+    const invalidInteractiveSyntaxes = [
+      { issue: 'use ', fix: "use('database')" },
+      { issue: 'show databases', fix: 'db.getMongo().getDBs()' },
+      { issue: 'show dbs', fix: 'db.getMongo().getDBs()' },
+      { issue: 'show collections', fix: 'db.getCollectionNames()' },
+      { issue: 'show tables', fix: 'db.getCollectionNames()' },
+      {
+        issue: 'show profile',
+        fix: "db.getCollection('system.profile').find()",
+      },
+      { issue: 'show users', fix: 'db.getUsers()' },
+      { issue: 'show roles', fix: 'db.getRoles({ showBuiltinRoles: true })' },
+      { issue: 'show logs', fix: "db.adminCommand({ getLog: '*' })" },
+      { issue: 'show log ', fix: "db.adminCommand({ getLog: 'global' })" },
+    ];
+
+    lines.forEach((line, i) => {
+      invalidInteractiveSyntaxes.forEach(({ issue, fix }) => {
+        const startCharacter = line.indexOf(issue);
+        let endCharacter = startCharacter + issue.length;
+
+        if (startCharacter >= 0) {
```

Comment:
Since this is a long conditional that wraps the rest of the function here, maybe it’s easier to do the inverse?

```suggestion
        if (startCharacter === -1) return;
```

(Also, totally random side note, but this PR adds both `for ... of` loops and `.forEach()` loops; might be nicer to use `for ... of` loops consistently? For the `lines.forEach((line, i) => {` one, you can use e.g. `for (const [ i, line ] of lines.entries()) {`)